### PR TITLE
Move the list of coercions to the reference

### PR DESCRIPTION
**Pull request by JohnTitor**
Helps #279:
- Removed the list of coercions and added a link to the reference (at a glance, the reference completely covers the items in the list here, so just removed)
- Tweaked the wording on some lines
- Updated the diagnostics

### CODE REVIEW

File: src/coercions.md [hunk 1]
```diff
@@ -1,58 +1,16 @@
...
-largely harmless.
+Types can implicitly be coerced to change in certain contexts.
+These changes are generally just *weakening* of types, largely focused around pointers and lifetimes.
+They mostly exist to make Rust "just work" in more cases, and are largely harmless.
 
-Here's all the kinds of coercion:
+For an exhaustive list of all the types of coercions, see the [Coercion types] section on the reference.
 
-Coercion is allowed between the following types:
+## Cases where coercions do not perform
```
Comment (Gankra) [inline src/coercions.md:9]:
```suggestion
## Cases where coercions do not happen
```

(but I would drop the subheading altogether -- this is already a very short page now)

File: src/coercions.md [hunk 2]
```diff
@@ -67,20 +25,21 @@ fn main() {
 }
 ```
 
+which fails like the following:
```
Comment (Gankra) [inline src/coercions.md:28]:
```suggestion
which fails as follows:
```

File: src/coercions.md [hunk 3]
```diff
@@ -1,58 +1,16 @@
...
-* Array literals: `let x: [U; 10] = [e, ..]`
-* Tuple literals: `let x: (U, ..) = (e, ..)`
-* The last expression in a block: `let x: U = { ..; e }`
-
-Note that we do not perform coercions when matching traits (except for
-receivers, see below). If there is an impl for some type `U` and `T` coerces to
-`U`, that does not constitute an implementation for `T`. For example, the
-following will not type check, even though it is OK to coerce `t` to `&T` and
-there is an impl for `&T`:
+Note that we do not perform coercions when matching traits (except for receivers, see below).
```
Comment (Gankra) [inline src/coercions.md:11]:
```suggestion
Note that we do not perform coercions when matching traits (except for receivers, see the next page).
```

This is a per-existing error, but still should be fixed.

**Review by Gankra — APPROVED** [review]
just a few edits

**Comment by JohnTitor** [comment]
Thanks! Applied all the review comments.

**Pull request merged**